### ISSUE TITLE
Duplicate Table Names Cannot Be Resolved

### ISSUE THREAD

**Issue by devinlyons** [fsprojects/SQLProvider#748]
**Describe the bug**
If you have a database with two tables named `TableName` and `Table_Name`, SQLProvider will be unable to resolve the names.

**To Reproduce**
1. Point SQLProvider at a database with pseudo duplicate names.
2. Access the tables in code and watch intellisense complain.

**Expected behavior**
SQLProvider should be able to resolve the names and display them in the list of tables.

**Screenshots**
![image](https://user-images.githubusercontent.com/8211199/134593361-eb2d2af7-5b99-459b-a279-dd3e5bc6e4f4.png)

**Desktop (please complete the following information):**
 - OS: Windows 10

**Comment by Thorium** [comment]
I'm not very happy with this fix as it breaks backward compatibility of existing system: The users probably have already table-names that have removed underscores in the tablename. SQLProvider is used already a lot in production systems.

So I'd prefer some kind of "if table name ends with underscore then don't remove it", fixing this special case only.

**Comment by halcwb** [comment]
> I'm not very happy with this fix as it breaks backward compatibility of existing system: The users probably have already table-names that have removed underscores in the tablename. SQLProvider is used already a lot in production systems.
> 
Indeed, I just wanted to show that a fix is possible. However, if underscores have been removed, this fix wouldn't be a problem any way. 

> So I'd prefer some kind of "if table name ends with underscore or table-name already exist, then don't remove it", fixing this special case only.
I would suggest making these kinds of naming strategies optional using additional optional constructor parameters. Then it would also be quite trivial to solve #759, without breaking backward compatibility.
